dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -191,11 +216,27 @@ func (rrh *resolverRequestHandler) RequestMiniBlock(destShardID uint32, minibloc
 	err = resolver.RequestDataFromHash(miniblockHash)
 	if err != nil {
 		log.Debug(err.Error())
+		return
+	}
+
+	err = rrh.requestedItemsHandler.Add(string(miniblockHash))
+	if err != nil {
+		log.Trace("add requested item with error",
+			"error", err.Error(),
+			"key", miniblockHash)
 	}
 }
 
 // RequestHeader method asks for header from the connected peers
 func (rrh *resolverRequestHandler) RequestHeader(destShardID uint32, hash []byte) {
+	rrh.requestedItemsHandler.Sweep()
```

Comment:
I do not think these sweeps are not needed.

Reply:
Refactored. I made an improvement to have sweep done only from time to time